src/Arrays/MPIStateArrays.jl

```diff
@@ -748,6 +748,8 @@ array_device(::Union{Array, SArray, MArray}) = CPU()
 array_device(::CuArray) = CUDADevice()
 array_device(s::SubArray) = array_device(parent(s))
 array_device(Q::MPIStateArray) = array_device(Q.data)
+array_device(ra::Base.ReshapedArray{T, N, A}) where {T, N, A <: MPIStateArray} =
+    array_device(ra.parent)
```

Comment:
```suggestion
    array_device(parent(ra))
```
Maybe? Not sure. `Base.ReshapedArray` is a `UnionAll`...

Reply:
That should work fine. I see `parent(A::ReshapedArray) = A.parent` in `julia/base/reshapedarray.jl`